datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/JwsAlgorithm.kt

```diff
@@ -20,77 +20,141 @@ enum class JwsAlgorithm(val identifier: String, override val oid: ObjectIdentifi
     ES256("ES256", KnownOIDs.ecdsaWithSHA256),
     ES384("ES384", KnownOIDs.ecdsaWithSHA384),
     ES512("ES512", KnownOIDs.ecdsaWithSHA512),
+
+    HS256("HS256", KnownOIDs.hmacWithSHA256),
+    HS384("HS384", KnownOIDs.hmacWithSHA384),
+    HS512("HS512", KnownOIDs.hmacWithSHA512),
+
+    PS256("PS256", KnownOIDs.rsaPSS),
+    PS384("PS384", KnownOIDs.rsaPSS),
+    PS512("PS512", KnownOIDs.rsaPSS),
+
     RS256("RS256", KnownOIDs.sha256WithRSAEncryption),
     RS384("RS384", KnownOIDs.sha384WithRSAEncryption),
     RS512("RS512", KnownOIDs.sha512WithRSAEncryption),
 
     /**
      * The one exception, which is not a valid JWS algorithm identifier
      */
-    NON_JWS_SHA1_WITH_RSA("RS1", KnownOIDs.sha1WithRSAEncryption),
-    HMAC256("HS256", KnownOIDs.hmacWithSHA256);
+    NON_JWS_SHA1_WITH_RSA("RS1", KnownOIDs.sha1WithRSAEncryption);
 
     val signatureValueLength
         get() = when (this) {
-            ES256 -> 256 / 8
-            ES384 -> 384 / 8
-            ES512 -> 512 / 8
-            HMAC256 -> 256 / 8
-            else -> -1 //TODO("RSA has no fixed size???")
+            ES256 -> 256 / 8 * 2
+            ES384 -> 384 / 8 * 2
+            ES512 -> 512 / 8 * 2
+            HS256 -> 256 / 8
+            HS384 -> 384 / 8
+            HS512 -> 512 / 8
+            else -> -1 // RSA signatures do not have a fixed size
```

Comment:
that special return type should be documented ... or return `null`?

Reply:
changed to null and commented